src/components/messageListener.ts

```diff
@@ -2,8 +2,36 @@ import { Message } from 'discord.js';
 import { applyBonusByUserId } from './coin';
 import { client } from '../bot';
 import { TextChannel } from 'discord.js';
+import { readFileSync } from 'fs';
+import { logError } from './logger';
+
+const ENV: string = process.env.NODE_ENV || 'dev';
+const vars = JSON.parse(readFileSync(`./config/${ENV}/vars.json`, 'utf-8'));
+const PC_CATEGORY_ID: string = vars.PC_CATEGORY_ID;
+const HONEYPOT_CHANNEL_ID: string = vars.HONEYPOT_CHANNEL_ID;
 
 export const messageListener = async (message: Message): Promise<void> => {
+  // Detects spammers/trolls/people who got hacked
+  const pingWords = ['@everyone', '@here'];
+  const punishableWords = ['http', 'nitro'];
+  if (
+    (pingWords.some((word) => message.content.includes(word)) &&
+      punishableWords.some((word) => message.content.toLowerCase().includes(word)) &&
+      message.channel instanceof TextChannel &&
+      message.channel.parentID !== PC_CATEGORY_ID) ||
+    message.channel.id === HONEYPOT_CHANNEL_ID
+  ) {
+    // Soft bans member and deletes messages from past 1 day
+    try {
+      if (await message.member?.ban({ days: 1, reason: 'Spammer/troll/got hacked' })) {
+        await message.guild?.members.unban(message.author.id);
```

Comment:
Just curious, is there a reason why we need to wrap this in an if statement, if `ban` fails, wouldn't it throw an error and skip `unban`? I.e. is there a case where `ban` returns `false` instead of throwing an error.

Reply:
In the if condition, `message.member` might be null, in which undefined would returned, which would evaluate to false. I guess in this case though, `message.member` would never be null based on the if statement above it, but I guess this *might* change in the future?

Reply:
Can you test out a code snippet that would trigger spam detection? like ```@everyone give nitro``` 
Also, plz ban for 1 days and 1 hour in case someone auto spams exactly and every 24 hours.
Do we have protection against an exec or mod or advisor or someone of somewhat higher CSC status accidently getting banned and all msgs deleted?

Reply:
Yup, the inline code block also triggers spam detection.
I think a soft ban in general is a decent punishment, since there's the added difficulty of having to rejoin the server before potentially spamming again. If a temporary ban would be implemented though, then would this involve storing the time of unban in the database and a cron job to check if it is time to do any unbans?
The Executive, Discord Mod, and Advisor roles are above the Codey role, so Codey can't possibly ban anyone with these roles.